sbi/inference/base.py

```diff
@@ -24,57 +25,56 @@ def __init__(
         simulator: Callable,
         prior,
         x_shape: Optional[torch.Size] = None,
+        num_workers: int = 1,
         simulation_batch_size: int = 1,
         retrain_from_scratch_each_round: bool = False,
-        device: Optional[torch.device] = None,
-        summary_writer: Optional[SummaryWriter] = None,
-        simulator_name: str = "simulator",
-        num_workers: int = 1,
         skip_input_checks: bool = False,
-        show_progressbar: bool = True,
+        exclude_invalid_x: bool = True,
+        device: Union[torch.device, str] = get_default_device(),
+        logging_level: Union[int, str] = "WARNING",
+        summary_writer: Optional[SummaryWriter] = None,
+        show_progress_bars: bool = True,
         show_round_summary: bool = False,
-        logging_level: Union[int, str] = "warning",
-        exclude_invalid_x: bool = False,
     ):
         r"""
+        Base class for inference algorithms.
+
         Args:
-            simulator: A regular callable $f(\theta)\to x$. Both parameter $\theta$ and
-                simulation $x$ can be multi-dimensional.
-            prior: Distribution-like object with `log_prob`and `sample` methods.
-            x_shape: Shape of a single simulation output $x$. Can be either (1, N) or
-                (N). Multidimensional observations (e.g. images) are not supported yet.
-                If None, the shape will be inferred by running a single simulation.
+            simulator: A function that takes parameters $\theta$ and maps them to
+                simulations, or observations, `x`, $\mathrm{sim}(\theta)\to x$. Any
+                regular Python callable (i.e. function or class with `__call__` method)
+                can be used.
+            prior: A probability distribution that expresses prior knowledge about the
+                parameters, e.g. which ranges are meaningful for them. Any
+                object with `.log_prob()`and `.sample()` (for example, a PyTorch
+                distribution) can be used.
+            x_shape: Shape of a single simulation output $x$, has to be (1,N).
+            exclude_invalid_x: Whether to exclude simulation outputs `x=NaN` or `x=±∞`
```

Comment:
- does not match the order of the arguments.
- missing `retrain_from_scratch_each_round`